base/parse.jl

```diff
@@ -131,19 +124,23 @@ function tryparse_internal{T<:Integer}(::Type{T}, s::AbstractString, startpos::I
     end
     return Nullable{T}(n)
 end
-tryparse_internal{T<:Integer}(::Type{T}, s::AbstractString, base::Int, raise::Bool) =
-    tryparse_internal(T,s,start(s),endof(s),base,raise)
-tryparse_internal{T<:Integer}(::Type{T}, s::AbstractString, startpos::Int, endpos::Int, base::Int, raise::Bool) =
-    tryparse_internal(T, s, startpos, endpos, base, base <= 36 ? 10 : 36, raise)
-tryparse{T<:Integer}(::Type{T}, s::AbstractString, base::Int) =
-    2 <= base <= 62 ? tryparse_internal(T,s,Int(base),false) : throw(ArgumentError("invalid base: base must be 2 ≤ base ≤ 62, got $base"))
-tryparse{T<:Integer}(::Type{T}, s::AbstractString) = tryparse_internal(T,s,0,false)
-
-function parse{T<:Integer}(::Type{T}, s::AbstractString, base::Integer)
-    (2 <= base <= 62) || throw(ArgumentError("invalid base: base must be 2 ≤ base ≤ 62, got $base"))
-    get(tryparse_internal(T, s, base, true))
+
+function tryparse_internal(::Type{Bool}, sbuff::AbstractString, startpos::Int, endpos::Int, base::Integer, raise::Bool)
+    len = endpos-startpos+1
+    p = pointer(sbuff)+startpos-1
+    (len == 4) && (0 == ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt), p, "true", 4)) && (return Nullable(true))
+    (len == 5) && (0 == ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt), p, "false", 5)) && (return Nullable(false))
```

Comment:
I don't think these are guaranteed to be valid on any arbitrary AbstractString subtype